### PR TITLE
Implemented pipeline. Connected the components

### PR DESCRIPTION
**Pull request by kaareHH**
The pipeline has been implemented, and can be run with the "mi_graph.py" file. 
Some of the components have been changed, in order to be connected with each other. 
Tests have been updated.

If you want to try and run the pipeline, please use the file in "examples/grundfos_test.json"

### CODE REVIEW

**Review by mkakbas — APPROVED** [review]

**Review by KasperSWinkler — DISMISSED** [review]

File: preprocess/cleaner_imp.py [hunk 1]
```diff
@@ -68,6 +83,12 @@ def lemmatize(self, words: str):
 
         return " ".join([token.lemma_ for token in doc])
 
+
+    def insert_pump_name(self, data, pump_name):
+        data = data.replace("the_pump", pump_name)
+        return data
+
```
Comment (AndersSpringborg) [inline preprocess/cleaner_imp.py:90]: there's no test on dis ting

File: tests/test_model.py [hunk 2]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
+        token = tokens[0].pop()
 
         assert token.pos_tag == GrammarCategories.noun
 
     def test_token_can_have_adj_label(self):
         self.model.load("good")
 
-        token = self.model.tokens().pop()
+        tokens = self.model.tokens()
+        token = tokens[0].pop()
```
Comment (AndersSpringborg) [inline tests/test_model.py:44]:
```suggestion
        first_sentance = self.model.tokens()[0]
        token =  first_sentance.pop()
```

File: tests/test_model.py [hunk 3]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
-        assert token in tokens
+        assert token in tokens[0]
 
     def test_token_has_label(self):
         self.model.load("apple")
 
-        token = self.model.tokens().pop()
+        tokens = self.model.tokens()
+        token = tokens[0].pop()
```
Comment (AndersSpringborg) [inline tests/test_model.py:36]:
```suggestion
        first_sentance= self.model.tokens()[0]
        token = first_sentance.pop()
```

File: tests/test_argsparser.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (AndersSpringborg) [inline tests/test_argsparser.py:50]:
```suggestion
```

delete lines

File: tests/test_model.py (hunk 3 above)
Comment (AndersSpringborg) [inline tests/test_model.py:36]:
```suggestion
        first_sentance= self.model.tokens()[0]
        token = first_sentance.pop()
```

File: tests/test_model.py (hunk 2 above)
Comment (AndersSpringborg) [inline tests/test_model.py:44]:
```suggestion
        first_sentance= self.model.tokens()[0]
        token = first_sentance.pop()
```

File: tests/test_model.py [hunk 5]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert token.pos_tag == GrammarCategories.adj
 
     def test_token_out_of_category(self):
         self.model.load("tyggegummi")
 
-        token = self.model.tokens().pop()
+        tokens = self.model.tokens()
+        token = tokens[0].pop()
```
Comment (AndersSpringborg) [inline tests/test_model.py:52]:
```suggestion
        first_sentance= self.model.tokens()[0]
        token = first_sentance.pop()
```

File: tests/test_model.py [hunk 6]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert token.pos_tag == GrammarCategories.other
 
     def test_detect_dependency_nsubj(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        apple = self.model.tokens()[0]
+        tokens = self.model.tokens()
+        apple = tokens[0][0]
```
Comment (AndersSpringborg) [inline tests/test_model.py:60]:
```suggestion
        first_sentance= self.model.tokens()[0]
        apple= first_sentance[0]
```

File: tests/test_model.py [hunk 7]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert apple.dep == Dependency.nsubj
 
     def test_detect_dependency_aux(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        _is = self.model.tokens()[1]
+        tokens = self.model.tokens()
+        _is = tokens[0][1]
```
Comment (AndersSpringborg) [inline tests/test_model.py:68]:
```suggestion
        first_sentance= self.model.tokens()[0]
        _is = first_sentance[1]
```

File: tests/test_model.py [hunk 8]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert _is.dep == Dependency.aux
 
     def test_detect_dependency_root(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        looking = self.model.tokens()[2]
+        tokens = self.model.tokens()
+        looking = tokens[0][2]
```
Comment (AndersSpringborg) [inline tests/test_model.py:76]:
```suggestion
        first_sentance= self.model.tokens()[0]
        looking = first_sentance[2]
```

File: tests/test_model.py [hunk 9]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert looking.dep == Dependency.root
 
     def test_detect_dependency_prep(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        at = self.model.tokens()[3]
+        tokens = self.model.tokens()
+        at = tokens[0][3]
```
Comment (AndersSpringborg) [inline tests/test_model.py:84]:
```suggestion
        first_sentance= self.model.tokens()[0]
        at = first_sentance[3]
```

File: tests/test_model.py [hunk 10]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert at.dep == Dependency.prep
 
     def test_detect_dependency_pcomp(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        buying = self.model.tokens()[4]
+        tokens = self.model.tokens()
+        buying = tokens[0][4]
```
Comment (AndersSpringborg) [inline tests/test_model.py:92]:
```suggestion
        first_sentance= self.model.tokens()[0]
        buying = first_sentance[4]
```

File: tests/test_model.py [hunk 11]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert buying.dep == Dependency.pcomp
 
     def test_detect_dependency_compund(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        UK = self.model.tokens()[5]
+        tokens = self.model.tokens()
+        UK = tokens[0][5]
```
Comment (AndersSpringborg) [inline tests/test_model.py:100]:
```suggestion
        first_sentance= self.model.tokens()[0]
        UK = first_sentance[5]
```

File: tests/test_model.py [hunk 12]
```diff
@@ -27,88 +27,128 @@ def test_input_sentence_in_tokens(self):
...
 
         assert UK.dep == Dependency.compound
 
     def test_detect_dependency_dobj(self):
         self.model.load("Apple is looking at buying U.K. startup for $1 billion")
 
-        startup = self.model.tokens()[6]
+        tokens = self.model.tokens()
+        startup = tokens[0][6]
```
Comment (AndersSpringborg) [inline tests/test_model.py:108]:
```suggestion
        first_sentance= self.model.tokens()[0]
        startup = first_sentance[6]
```

**Review by AndersSpringborg — CHANGES_REQUESTED** [review]

**Review by AndersSpringborg — APPROVED** [review]
Very nice

**Review by mkakbas — APPROVED** [review]

**Pull request merged**